### ISSUE TITLE
Enable usage of openshift_form_definition in schema definition

### ISSUE THREAD

**Issue by pgrabarsky** [spring-cloud/spring-cloud-cloudfoundry-service-broker#61]
OpenShift enable to specify in which order the parameters appear into their GUI by using the convention described here: https://github.com/openshift/origin/issues/16383. By default the GUI is showing parameters sorted by their name (actually the GUI is inline with the ClusterServicePlan and it is this last one that is sorting...).
Currently, the MethodSchema class only support the 'parameters' attribute, hence preventing from using this feature. 
A possible solution to enable the usage of this OpenShift feature would be to enable to MethodSchema to accept more attributes or to find a way to enable model customization.

**Comment by scottfrederick** [comment]
@pgrabarsky This project attempts to strictly adhere to the [Open Service Broker API spec](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md). There are several places in the spec that allow for arbitrary objects in requests and responses to account for platform-specific features (e.g. [`metadata` fields in the catalog](https://github.com/openservicebrokerapi/servicebroker/blob/master/profile.md#service-metadata)). 

The [specification for schemas](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#schema-object) currently lists only `parameters` as a valid field in schema objects. I'd be more comfortable adding support like you're suggesting to this project if the spec suggested a uniform way for such platform-specific features to be added. 

I don't see an [open issue](https://github.com/openservicebrokerapi/servicebroker/issues) related to adding support for arbitrary additional fields to schemas. Do you have any visibility into whether the OpenShift or any other teams has proposed a change?

**Comment by pgrabarsky** [comment]
@scottfrederick Actually, I had actually overlooked the PR description. The impacted section is inside the metadata part so there is no need to change this project behavior to handle it. I'm sorry for that.

BTW, for the ones that would have to handle the case and that would read this ticket, I've created a gist that generated automatically this piece of data at serialization time: https://gist.github.com/pgrabarsky/cdbe0c7c8bd8ccc0b69e2529ad21d3a5 The outcome of it, is that you have nice automated GUI generation inside OpenShift.

And finally, to answer your question, I've seen anything like real extension point into the spec and I don't know the details of the current discussion. But, I hope the API will evolve as I find it a bit limited to handle complex uses cases.
